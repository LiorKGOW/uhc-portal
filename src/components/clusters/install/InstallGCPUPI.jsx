import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { PageSection } from '@patternfly/react-core';

import { AppPage } from '~/components/App/AppPage';

import { tollboothActions } from '../../../redux/actions';
import Breadcrumbs from '../../common/Breadcrumbs';
import PageTitle from '../../common/PageTitle';

import instructionsMapping from './instructions/instructionsMapping';
import OCPInstructions from './instructions/OCPInstructions';

export class InstallGCPUPI extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const breadcrumbs = (
      <Breadcrumbs
        path={[
          { label: 'Clusters' },
          { label: 'Cluster Type', path: '/create' },
          { label: 'Google Cloud Platform', path: '/install/gcp' },
          { label: 'User-provisioned infrastructure' },
        ]}
      />
    );

    return (
      <AppPage title="Install OpenShift 4 | Red Hat OpenShift Cluster Manager | GCP User-Provisioned Infrastructure">
        <PageTitle title={instructionsMapping.gcp.upi.title} breadcrumbs={breadcrumbs} />
        <PageSection>
          <OCPInstructions
            token={token}
            cloudProviderID="gcp"
            isUPI
            {...instructionsMapping.gcp.upi}
          />
        </PageSection>
      </AppPage>
    );
  }
}

InstallGCPUPI.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({ token: state.tollbooth.token });

export default connect(mapStateToProps)(InstallGCPUPI);
