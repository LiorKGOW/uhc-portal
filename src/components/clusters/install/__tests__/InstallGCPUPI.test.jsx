import React from 'react';
import { shallow } from 'enzyme';

import { InstallGCPUPI } from '../InstallGCPUPI';

describe('InstallGCPUPI', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InstallGCPUPI token={{}} dispatch={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
