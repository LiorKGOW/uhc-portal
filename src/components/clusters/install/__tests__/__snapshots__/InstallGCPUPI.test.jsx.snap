// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallGCPUPI renders correctly 1`] = `
<Fragment>
  <PageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Create",
              "path": "/create",
            },
            Object {
              "label": "Google Cloud Platform",
              "path": "/install/gcp",
            },
            Object {
              "label": "User-provisioned infrastructure",
            },
          ]
        }
      />
    }
    title="Install OpenShift on GCP with user-provisioned infrastructure"
  />
  <PageSection>
    <OCPInstructions
      channel="stable"
      cloudProviderID="gcp"
      customizations="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-customizations.html"
      displayRHCOSSection={true}
      docURL="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-user-infra.html"
      installer="x86_64-openshift-install"
      isBMIPI={false}
      rhcosDownloads={
        Array [
          Object {
            "buttonText": "Download RHCOS tar",
            "name": "OCP-Download-RHCOS-tar",
            "url": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/latest/rhcos-4.5.6-x86_64-gcp.x86_64.tar.gz",
          },
        ]
      }
      rhcosLearnMoreURL="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-user-infra.html#installation-gcp-user-infra-rhcos_installing-gcp-user-infra"
      title="Install OpenShift on GCP with user-provisioned infrastructure"
      token={Object {}}
    />
  </PageSection>
</Fragment>
`;
